recorder.go

```diff
@@ -420,28 +452,20 @@ func (r *Recorder) Flush() {
 		return
 	}
 
+	// not in flight => r.flushing has been reset
```

Comment:
Updated. Just to say that r.flushing must have been reset and is therefore ready to use.